tests/inference_on_device_test.py

```diff
@@ -18,30 +26,31 @@
 @pytest.mark.parametrize(
     "method, model",
     [
-        (SNPE, "mdn"),
-        (SNPE, "maf"),
-        (SNPE, "nsf"),
+        (SNPE_A, "mdn_snpe_a"),
+        (SNPE_C, "mdn"),
+        (SNPE_C, "maf"),
         (SNLE, "maf"),
         (SNLE, "nsf"),
         (SNRE_A, "mlp"),
         (SNRE_B, "resnet"),
     ],
 )
-@pytest.mark.parametrize("device", ("cpu", "cuda:0"))
-def test_training_and_mcmc_on_device(method, model, device):
+@pytest.mark.parametrize("data_device", ("cpu", "cuda:0"))
+@pytest.mark.parametrize("training_device", ("cpu", "cuda:0"))
+def test_training_and_mcmc_on_device(method, model, data_device, training_device):
     """Test training on devices.
 
     This test does not check training speeds.
 
     """
-    device = process_device(device)
+    training_device = process_device(training_device)
 
     num_dim = 2
     num_samples = 10
     num_simulations = 500
     max_num_epochs = 5
 
-    x_o = zeros(1, num_dim)
+    x_o = zeros(1, num_dim).to(data_device)
```

Comment:
Is this necessary now? Only necessary for SNPE-A?

Reply:
No, this change was made to also test for the case where the data is already on the device, replacing this test: https://github.com/mackelab/sbi/blob/9d321baaa136b6901d3db17e7c36f102e274e6fe/tests/user_input_checks_test.py#L513-L549